Add snap and lock angle modifiers for handle dragging to the Path tool

bakayu opened the pull request:
Closes #1870 partly...
This PR aims to close the following parts of #1870 :
> - [x] <kbd>Shift</kbd> locks the handle angle to 15° increments
>   - [x] (shouldn't interfere with <kbd>Shift</kbd>-clicking handles for multi-selecting points such as handles)
> - [x] <kbd>Ctrl</kbd> locks the current angle of the handle

Keavon commented:
!build

bakayu commented:
@Keavon This PR is now ready for review!
Please produce a dev build for testing ^^

Keavon commented:
!build

Keavon commented:
Please check off the boxes in the issue description for what's implemented now with this PR.

bakayu commented:
By now, this PR has implemented all of the listed features. Updated the description as well

Keavon commented:
QA feedback:
- If the handles are not colinear, releasing <kbd>Alt</kbd> should put the opposite handle back where it started before Alt was pressed so the user can have the freedom to press <kbd>Alt</kbd> any number of times without worrying about affecting things by changing your mind and releasing <kbd>Alt</kbd.

bakayu commented:
> QA feedback:
> 
> * If the handles are not colinear, releasing Alt should put the opposite handle back where it started before Alt was pressed so the user can have the freedom to press Alt any number of times without worrying about affecting things by changing your mind and releasing Alt.

Noted.
For now, I have reverted the commit that aimed to fix this, I will address this later in a seperate PR. 

Updated the description to reflect what this PR currently solves.

Keavon commented:
!build

Keavon commented:
!build

Keavon commented:
There's one more issue I found, but it can be fixed in the next PR you work on:
- If you begin dragging a handle with <kbd>Ctrl</kbd> already pressed, it uses the angle of the *previously selected handle* rather than the being-dragged handle.

bakayu commented:
I will continue the work in a couple days, will fix it then.

Keavon commented:
Oh and one more:
- If I'm holding <kbd>Ctrl</kbd> to lock the angle, then add <kbd>Shift</kbd> to snap it to 15°, then release <kbd>Shift</kbd>, it should return to its previously locked angle that was set when the currently held-down <kbd>Ctrl</kbd> was pressed down. This also needs to be true for the Line tool, which isn't currently the case. Also for the Pen tool, which currently doesn't even let you add <kbd>Shift</kbd> while <kbd>Ctrl</kbd> is held. What I'm describing should be its own PR.
